dotCMS/src/main/java/com/dotcms/util/VanityUrlUtil.java

```diff
@@ -0,0 +1,93 @@
+package com.dotcms.util;
+
+import com.dotcms.contenttype.model.type.VanityUrlContentType;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.HostAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+
+/**
+ * This class provide some utility methods to interact with
+ * Vanity URLs
+ *
+ * @author oswaldogallango
+ * @version 4.2.0
+ * @since June 12, 2017
+ */
+public class VanityUrlUtil {
+
+    private static final HostAPI hostAPI = APILocator.getHostAPI();
+
+    private VanityUrlUtil() {
+
+    }
+    /**
+     * Generate the sanitized cache key
+     *
+     * @param hostId The current host Identifier
+     * @param uri The current URI
+     * @param languageId The current languageId
+     * @return String with the sanitized key name
+     */
+    public static String sanitizeKey(final String hostId, final String uri, final long languageId) {
+        return hostId + "|" + fixURI(uri).replace('/', '|') + "|lang_" + languageId;
+    }
+
+    /**
+     * Generate the sanitized cache key
+     *
+     * @param vanityUrl The vanity Url contentlet
+     * @return String with the sanitized key name
+     */
+    public static String sanitizeKey(final Contentlet vanityUrl)
+            throws DotDataException, DotRuntimeException, DotSecurityException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)